### PR TITLE
Fix RFE-4185: run bundle-upgrade with long image names

### LINKED ISSUE #6476: Fail to run bundle-upgrade if the bundle image names are too long
## Bug Report

In CI system, the bundle image names may be very long. In this case `operator-sdk run bundle-upgrade` is failing.

#### What did you do?
1. Installed an "old" bundle 
2. Ran `operator-sdk run bundle-upgrade` with the new bundle that created in ci

#### What did you expect to see?
Upgrade is done successfully.

#### What did you see instead? Under which circumstances?
the operator SDK fails with the following message:

```
level=fatal msg="Failed to run bundle upgrade: update catalog error: error in upgrading FBC: error creating the upgraded FBC: error creating new image registry: mkdir registry.build03.ci.openshift.org-ci-op-dl9pitlh-pipeline@sha256:123d009f7fb9d5b760b3eac831e0f50413d36199816d651b569367642c138ccd_registry.build03.ci.openshift.org-ci-op-dl9pitlh-pipeline@sha256:5498f1b841946c8da823fb11c08969efb36119f7867701f2cedac4a2e8f7cefc: file name too long"
```

#### Environment

**Operator type:**

**Kubernetes cluster type:**

`$ operator-sdk version` = v1.29.0

`$ go version` (if language is Go)

`$ kubectl version`

#### Possible Solutions
1. Hash the long directory name, to make it shorter, but still unique; OR
2. Create a temporary directory using `os.TempDir()`

### PR DESCRIPTION
**Pull request by nunnatsa**
Fix #6476 

## Description of the change
In `run bundle-upgrade`, hash the cache directory name to avoid error of too long file name.

## Motivation for the change
`run bundle-upgrade` is failing if the bundle image names are long. See #6476 

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: internal/olm/fbcutil/util_test.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	Context("test dirNameFromRefs", func() {
+		DescribeTable("should name the directory with sha256 of the concatenation of the bundle image names", func(refs []string, expectedHash string) {
+			dirName := dirNameFromRefs(refs)
+
+			Expect(dirName).Should(HaveLen(sha256.Size * 2)) // in hex representation, each byte is two hex digits
+			Expect(dirName).Should(Equal(expectedHash))
+		},
+			Entry("long image names", []string{
+				"registry.build03.ci.openshift.org/ci/op/dl9pitlh/pipeline@sha256:123d009f7fb9d5b760b3eac831e0f50413d36199816d651b569367642c138ccd",
+				"registry.build03.ci.openshift.org/ci/op/dl9pitlh/pipeline@sha256:5498f1b841946c8da823fb11c08969efb36119f7867701f2cedac4a2e8f7cefc",
```
Comment (everettraven) [inline internal/olm/fbcutil/util_test.go:39]: Might be nice to use non-OpenShift references here. I imagine we can use any long string here

File: internal/olm/fbcutil/util.go [hunk 2]
```diff
@@ -169,6 +171,14 @@ func RenderRefs(ctx context.Context, refs []string, skipTLSVerify bool, useHTTP
 	return declcfg, nil
 }
 
+func dirNameFromRefs(refs []string) string {
+	dirNameBytes := []byte(strings.ReplaceAll(strings.Join(refs, "_"), "/", "-"))
+	hash := sha256.New()
+	hash.Write(dirNameBytes)
+	hashBytes := hash.Sum(nil)
+	return fmt.Sprintf("%x", hashBytes)
```
Comment (everettraven) [inline internal/olm/fbcutil/util.go:179]: Can we shorten this to something like:
```suggestion
	return string(hash.Sum(nil))
```
?

If not this is fine, but I think those are equivalent and doesn't require use of the `fmt.Sprintf` function

**Review by everettraven — APPROVED** [review]
Overall looks good to me. Just a couple nits (non-blocking IMO but would be nice to have addressed)

File: internal/olm/fbcutil/util.go (hunk 2 above)
Comment (nunnatsa) [reply internal/olm/fbcutil/util.go:179]: Thanks for the comment @everettraven - but these two forms are not equivalent. `string(hash.Sum(nil))` will try to use the random bytes as characters. The result is unexpected - we'll end up with wrong UTF-8 chars, non-printable chars and so on. 

The original format generates a hex string (2 hex digit per byte).
